### ISSUE TITLE
Implement approximate ShapValues calculation

### ISSUE THREAD

**Issue by victorbartel** [catboost/catboost#1146]
Problem: `get_feature_importance` with `fstr_type='ShapValues'` has poor performances
catboost version: _0.20.2_
Operating System: _Ubuntu 19.10_
CPU: _Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz_
model type: Regression 
model format: cbm
model size: ~250mb

Greetings,

First of all, thanks a lot for such a great library. I'm facing some issues with explanation of predicted values. While the prediction works blazingly fast the explanation of predictions through shap-values shows poor performances.

In order to understand my issue here are few figures:
Prediction time: ~ _0.0099 sec_
Shap values time: ~ _0.7799 sec_

Here is a portion of my code:
```python
features = MODEL.get_feature_importance(data=pool, fstr_type='ShapValues')
shap_values, base_value = np.split(features[0], [-1])
```

My primary goal is to explain the prediction for a given item at the same time when prediction is made. Ideally I would like to launch everything inside of a GCP cloud-function.

Many thanks.

**Comment by annaveronika** [comment]
This time is completely fine, because shap values calculation is a much more expensive procedure than prediction. Here is the time complexity of shap values: https://github.com/catboost/benchmarks/tree/master/shap_speed

**Comment by victorbartel** [comment]
Hey Anna-Veronika,

Thanks a lot, I got the point. I saw as well that in shap in `TreeExplainer` we can use `shap_values` method with an approximation flag ([approximate](https://shap.readthedocs.io/en/latest/#shap.TreeExplainer.shap_values)) which is not implemented yet by `CatBoost`. Do you have any plans to implement it in the nearest future and could such approximation improve shap value generation(obviously by sacrificing its precision)?

Many thanks.

**Comment by annaveronika** [comment]
This is a great idea, we'll implement this, thanks a lot for the suggestion!

**Comment by andrey-khropov** [comment]
This functionality has been implemented and fully works since commit 13d0a64cb0c1d0bbc49a9178ab17f53dcae140c5 and will be included in the next release.

Specify `shap_calc_type` parameter of  `CatBoost.get_feature_importance` function as  `"Approximate"` to enable it (it is `"Normal"` by default).
